Mathlib/Algebra/CharP/Basic.lean

```diff
@@ -560,7 +560,20 @@ end Semiring
 
 section Ring
 
-variable [Ring R] [NoZeroDivisors R] [Nontrivial R] [Finite R]
+variable [Ring R] [Nontrivial R]
+
+theorem charP_iff_eq_zero_of_prime {p : ℕ} (hp : p.Prime) : CharP R p ↔ (p : R) = 0 :=
+  ⟨fun h => cast_eq_zero R p, by
+    rw [charP_iff]
+    intro hp0 n
+    refine ⟨?_, fun ⟨_, hpn⟩ => by simp_all⟩
+    rw [hp.dvd_iff_not_coprime, Nat.coprime_iff_gcd_eq_one,
+      ← Nat.cast_inj (R := ℤ), Nat.gcd_eq_gcd_ab]
+    intro hn0 hgcd
+    apply_fun ((↑) : ℤ → R) at hgcd
+    simp [hp0, hn0] at hgcd⟩
```

Comment:
Is there a faster route here via `ringChar_of_prime_eq_zero` (which is incorrectly in the `CharOne` section below)

Reply:
Done and I also renamed the section